signature/algorithm.go

```diff
@@ -0,0 +1,159 @@
+package signature
```

Comment:
Should this file go to the `plugin` package since they are all plugin protocol related stuffs?

Reply:
Updated